editor/src/components/canvas/canvas-strategies/strategies/grid-cell-bounds.ts

```diff
@@ -27,6 +33,38 @@ export function gridCellTargetId(
   return gridCellTargetIdPrefix + `${EP.toString(gridElementPath)}-${row}-${column}`
 }
 
+export function getGridCellUnderMouseFromMetadata(
+  grid: ElementInstanceMetadata,
+  canvasPoint: CanvasPoint,
+): TargetGridCellDataCanvas | null {
+  const gridCellGlobalFrames = getGlobalFramesOfGridCells(grid)
+
+  if (gridCellGlobalFrames == null) {
+    return null
+  }
+
+  return getGridCellUnderPoint(gridCellGlobalFrames, canvasPoint)
+}
+
+// TODO: optimization: binary search
+function getGridCellUnderPoint(
+  gridCellGlobalFrames: GridCellGlobalFrames,
+  canvasPoint: CanvasPoint,
+): TargetGridCellDataCanvas | null {
+  for (let i = 0; i < gridCellGlobalFrames.length; i++) {
+    for (let j = 0; j < gridCellGlobalFrames[i].length; j++) {
+      if (rectContainsPoint(gridCellGlobalFrames[i][j], canvasPoint)) {
+        return {
+          gridCellCoordinates: gridCellCoordinates(i + 1, j + 1),
+          cellCanvasRectangle: gridCellGlobalFrames[i][j],
+        }
+      }
+    }
+  }
+  return null
+}
+
+// TODO: should be superseded by getGridCellUnderMouseNoDomLookup
```

Comment:
do you mean

```suggestion
// TODO: should be superseded by getGridCellUnderMouseFromMetadata
```


?

Reply:
yes, meanwhile I renamed the function :)